plugins/libplugin.c

```diff
@@ -710,11 +719,37 @@ static struct io_plan *rpc_conn_init(struct io_conn *conn,
 			 rpc_conn_write_request(conn, plugin));
 }
 
+static struct feature_set *json_to_feature_set(struct plugin *plugin,
+					       const char *buf,
+					       const jsmntok_t *features)
+{
+	struct feature_set *fset = talz(plugin, struct feature_set);
+	const jsmntok_t *t;
+	size_t i;
+
+	json_for_each_obj(i, t, features) {
+		enum feature_place p;
+		if (json_tok_streq(buf, t, "init"))
+			p = INIT_FEATURE;
+		else if (json_tok_streq(buf, t, "node"))
+			p = NODE_ANNOUNCE_FEATURE;
+		else if (json_tok_streq(buf, t, "channel"))
+			p = CHANNEL_FEATURE;
+		else if (json_tok_streq(buf, t, "invoice"))
+			p = BOLT11_FEATURE;
+		else
+			continue;
```

Comment:
This should likely at least produce a warning that the plugin is not in sync with `lightningd`

Reply:
I figure it's always legal to add  a field.  Though perhaps libplugin should consider itself more tightly bound?